src/input/src/simulations/logging.rs

```diff
@@ -31,12 +31,12 @@ impl Encode for LogEncoder {
             LogLevel::Debug => try!(write!(out, "[debug] ")),
             LogLevel::Info => {},
             LogLevel::Warn => {
-                try!(out.set_style(&Style::new().text(Color::Red)));
+                try!(out.set_style(Style::new().text(Color::Red)));
```

Comment:
I don't get what happens here, you can just substitute a reference for a plain struct ?